server/service/http/message/query/row.rs

```diff
@@ -10,14 +10,22 @@ use answer::variable_value::VariableValue;
 use compiler::VariablePosition;
 use concept::{error::ConceptReadError, thing::thing_manager::ThingManager, type_::type_manager::TypeManager};
 use executor::row::MaybeOwnedRow;
+use serde::Serialize;
 use serde_json::json;
 use storage::snapshot::ReadableSnapshot;
 
 use crate::service::http::message::query::concept::{encode_thing_concept, encode_type_concept, encode_value};
 
-pub fn encode_row(
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct EncodedRow<'a> {
+    data: HashMap<&'a str, serde_json::Value>,
+    provenance: u64,
```

Comment:
I wonder if we should encode this as a set of Integers that are indexes into the Branches in the QueryStructure - right now we're coupling our server-side compact encoding of Provenance onto the client-side, which is not ideal IMO.

The cost is going to be some more space... but i think this might be ok to avoid this coupling.